### PR TITLE
Recommended section version

### PR DESCRIPTION
**Pull request by maddiedierker**
[LP-28](https://codedotorg.atlassian.net/browse/LP-28)
[Spec](https://docs.google.com/document/d/1jYznJYlrfXQ-MJvmgfmZbVjjAM9-i297hoFTCqoYJ5w/edit?usp=sharing)

Satisfies the following requirements in the above spec:
- If you’re in a language that is supported by a stable version of a course, we should show that as the recommended version for that course.
- If my selected language is not supported by any of the versions, the recommended version should continue to be the latest stable version in English. For example, regardless of what language you’re in for CSP, the recommended version will currently be 2018 (as of Feb 2019). 
- Set the recommended version as the dropdown default (no additional work was done to make this happen, just calling it out here)

### Example
Let's say our locale language is Spanish. `coursea-2018` does not support Spanish, but `coursea-2017` does.

#### Before
We always displayed the latest stable version as the recommended version:
Screen Shot 2019-04-09 at 3 19 39 PM

#### After
We default to and recommend the latest stable version in the user's locale. If no versions are supported in their locale, we recommend the latest stable version.

Default in dropdown:
Screen Shot 2019-04-09 at 3 16 18 PM

Upon opening dropdown:
Screen Shot 2019-04-09 at 3 16 28 PM

### CODE REVIEW

File: dashboard/app/views/home/_homepage.html.haml [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - homepage_data[:stageExtrasScriptIds] = Script.stage_extras_script_ids
 - homepage_data[:topCourse] = @top_course
 - homepage_data[:isEnglish] = @is_english
+- homepage_data[:locale] = Script.locale_english_name_map[request.locale]
```
Comment (davidsbailey) [inline dashboard/app/views/home/_homepage.html.haml:10]: another gripe that I'm not sure we should do anything about: I really want to move this to `LocaleHelper` module, but the method caches internally via class variables and I'm not sure you can set class variables inside a module.

**Review by davidsbailey — APPROVED** [review]
Hi Maddie, some things are setting off minor alarm bells here:
* comparing translated strings rather than the more reliable locale short codes 
* passing the `locale` piece of data all the way down to a react subcomponent on the client when the same calculation for recommended version could be done on the server

Unfortunately, there's a bit of other should-be-server-side logic in the client in a couple places which makes this non-trivial to fix this:
* computing recommended version in  `AssignmentSelector#getVersions`
* computing family groupings in teacherSectionsRedux

As for the UI string, while comparing to a well-defined shortcode seems better, I'm not convinced it's worth passing even more data down to the client to avoid comparing UI strings.

So, I'm inclined to say let's go with how you have it, and some day we can think about moving family groupings and recommended version detection to the server. If we do that, then the UI string comparison will go away too.

File: dashboard/app/views/home/_homepage.html.haml (hunk 1 above)
Comment (maddiedierker) [reply dashboard/app/views/home/_homepage.html.haml:10]: yeah, i had the same feeling. unfortunately, we can't use a module, but we could create a singleton class that holds and caches this data. that's very "un-rails," so i'm not sure what the best way to go about that would be

**Comment by maddiedierker** [comment]
@davidsbailey thanks! i think your concerns are valid, but i agree that it might be too tricky to do the work right now. 

re: your comment...
> passing the locale piece of data all the way down to a react subcomponent on the client when the same calculation for recommended version could be done on the server

i considered creating a redux store to store the user's locale (there isn't an existing store where it really makes sense) to avoid passing the locale through lots of components. do you think that's a valuable refactor to make right now?

**Comment by davidsbailey** [comment]
Good idea, but I'm not sure that the immediate payoff will be great, because we currently plug redux into a react component near the top level, and then pass it down through many subcomponents anyway. so I think it is fine to do it this way and say that the long term plan would be to move this logic to the server.

**Pull request merged**